kai/reactive_codeplanner/agent/analyzer_fix/agent.py

```diff
@@ -22,30 +22,27 @@
 @dataclass
 class _llm_response:
     reasoning: str | None
-    java_file: str | None
+    source_file: str | None
     addional_information: str | None
 
 
 class AnalyzerAgent(Agent):
-    system_message = SystemMessage(
-        content="""
-    You are an experienced java developer, who specializes in migrating code to the Quarkus Framework
+    system_message_template = Template(
+        """
+    You are an experienced {{ language }} developer, who specializes in migrating code from {{ source }} to {{ target }}
```

Comment:
Currently I'm getting really awesome high quality prompts that look like this:


```
You are an experienced java developer, who specializes in migrating code from  to eap8 and eap and jws6+ and hibernate6+ and hibernate and eap8+ and quarkus3+ and jakarta-ee and jws and jakarta-ee9+ and quarkus
    You are a software developer who specializes in migrating from  to eap8 and eap and jws6+ and hibernate6+ and hibernate and eap8+ and quarkus3+ and jakarta-ee and jws and jakarta-ee9+ and quarkus
You attempted to solve an issue in a repository you are migrating:

Location: file:///example/coolstore/src/main/java/com/redhat/coolstore/model/Order.java
Message:
The way in which Hibernate determines implicit names for sequences and tables associated with identifier generation has changed in 6.0 which may affect migrating applications. 
 As of 6.0, Hibernate by default creates a sequence per entity hierarchy instead of a single sequence hibernate_sequence. 
 Due to this change, users that previously used `@GeneratedValue(strategy = GenerationStrategy.AUTO)` or simply `@GeneratedValue` (since `AUTO` is the default), need to ensure that the database now contains sequences for every entity, named `<entity name>_seq`. For an entity Person, a sequence person_seq is expected to exist. 
 It’s best to run hbm2ddl (e.g. by temporarily setting `hbm2ddl.auto=create`) to obtain a list of DDL statements for the sequences.

However your solution caused additional problems elsewhere in the repository, which you are now going to solve.
```

and 

```
You are an experienced xml developer, who specializes in migrating code from java-ee and java to cloud-readiness
    None
```

Reply:
So refinement is needed to say the least

Reply:
Also we potentially may want to add the TaskResult objects to the Tasks, which would allow us to include some specific information about the changes made prior in the prompt as well (perhaps it would also make sense to generate and attach the diff in the TaskResult?)